modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb

```diff
@@ -24,67 +24,67 @@ def show
           end
 
-          submit_power_of_attorney(poa_code)
+          submit_power_of_attorney(poa_code, 'submit_2122')
         end
 
-        def appoint_individual
+        def submit_2122a
+          shared_2122a_validation('representative')
+
           validate_request!(ClaimsApi::V2::ParamsValidation::PowerOfAttorney)
           poa_code = parse_and_validate_poa_code
           if poa_code_in_organization?(poa_code)
             raise ::Common::Exceptions::UnprocessableEntity.new(detail: 'POA Code must belong to an individual.')
           end
 
-          submit_power_of_attorney(poa_code)
+          submit_power_of_attorney(poa_code, 'submit_2122a')
         end
 
-        def submit_power_of_attorney(poa_code)
-          power_of_attorney = ClaimsApi::PowerOfAttorney.find_using_identifier_and_source(header_md5:,
-                                                                                          source_name:)
-          unless power_of_attorney&.status&.in?(%w[submitted pending])
-            attributes = {
-              status: ClaimsApi::PowerOfAttorney::PENDING,
-              auth_headers:,
-              form_data: params,
-              current_poa: current_poa_code,
-              header_md5:
-            }
-            attributes.merge!({ source_data: }) unless token.client_credentials_token?
+        def validate_2122a
+          shared_2122a_validation('representative')
 
-            # use .create! so we don't need to check if it's persisted just to call save (compare w/ v1)
-            power_of_attorney = ClaimsApi::PowerOfAttorney.create!(attributes)
-          end
+          render json: validation_success
+        end
+
+        private
 
-          ClaimsApi::PoaUpdater.perform_async(power_of_attorney.id)
+        def submit_power_of_attorney(poa_code, action)
+          attributes = {
+            status: ClaimsApi::PowerOfAttorney::PENDING,
+            auth_headers:,
+            form_data: form_attributes,
+            current_poa: current_poa_code,
+            header_md5:
+          }
+          attributes.merge!({ source_data: }) unless token.client_credentials_token?
 
-          ClaimsApi::VBMSUpdater.perform_async(power_of_attorney.id) if enable_vbms_access?
+          power_of_attorney = ClaimsApi::PowerOfAttorney.create!(attributes)
 
           # This builds the POA form *AND* uploads it to VBMS
           ClaimsApi::PoaFormBuilderJob.perform_async(power_of_attorney.id)
 
           render json: ClaimsApi::V2::Blueprints::PowerOfAttorneyBlueprint.render(
-            representative(poa_code).merge({ code: poa_code })
+            representative(poa_code).merge({ id: power_of_attorney.id, code: poa_code }),
+            root: :data
+          ), status: :accepted, location: url_for(
+            controller: 'power_of_attorney', action:, id: power_of_attorney.id
           )
         end
 
-        def validate2122a
+        def shared_2122a_validation(rep_or_org)
```

Comment:
I like this suggestion. I should point out that this shared validation is going to change soonish with the signatures work because 2122 won't care about registration numbers, but 2122a will.